### PR TITLE
Port new template structure to the tool

### PR DESCRIPTION
**Pull request by AnjanaSenanayake**
Port new template structure to the tool

Related Issue: https://github.com/wso2-enterprise/open-healthcare/issues/1400

### CODE REVIEW

File: native/health-cli/src/main/java/io/ballerina/health/cmd/Runner.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                targetOutputPath = Paths.get(outputPath);
+            } else {
+                targetOutputPath = Paths.get(targetOutputPath.toString(), outputPath);
+            }
+        } else {
+            targetOutputPath = Paths.get(targetOutputPath + File.separator + "generated");
+        }
+        return targetOutputPath;
+    }
+}
```
Comment (isuruh15) [inline native/health-cli/src/main/java/io/ballerina/health/cmd/Runner.java:87]: Check EoF newlines

File: native/health-cli/src/main/java/io/ballerina/health/cmd/fhir/FhirSubCmd.java [hunk 2]
```diff
@@ -84,9 +84,12 @@ public class FhirSubCmd implements BLauncherCmd {
     @CommandLine.Option(names = "--excluded-profile", description = "Profiles to be excluded in the template")
     private String[] excludedProfiles;
 
+    @CommandLine.Option(names = "--resourcePackage", description = "Resource package name for the templates to be generated")
```
Comment (isuruh15) [inline native/health-cli/src/main/java/io/ballerina/health/cmd/fhir/FhirSubCmd.java:87]: let's use dash delimited strings when defining the named arguments.
`--resourcePackage` -> `--resource-package`

File: native/fhir-to-bal-template/src/main/java/org/wso2/healthcare/fhir/codegen/ballerina/project/tool/generator/ServiceGenerator.java [hunk 3]
```diff
@@ -29,9 +29,7 @@
 import org.wso2.healthcare.fhir.codegen.ballerina.project.tool.model.ResourceMethod;
 
 import java.io.File;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (AnjanaSenanayake) [reply native/fhir-to-bal-template/src/main/java/org/wso2/healthcare/fhir/codegen/ballerina/project/tool/generator/ServiceGenerator.java]: Fixed in https://github.com/ballerina-platform/fhir-tools/commit/233fe707741f4680c2b3215341dc823046c19930

File: native/health-cli/src/main/java/io/ballerina/health/cmd/Runner.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import io.ballerina.health.cmd.core.utils.HealthCmdUtils;
+import io.ballerina.health.cmd.handler.Handler;
+import io.ballerina.health.cmd.handler.HandlerFactory;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
```
[the remark this answers is not in the document]
Comment (AnjanaSenanayake) [reply native/health-cli/src/main/java/io/ballerina/health/cmd/Runner.java]: This was added by a mistake. Removed in https://github.com/ballerina-platform/fhir-tools/commit/233fe707741f4680c2b3215341dc823046c19930

File: native/health-cli/src/main/java/io/ballerina/health/cmd/Runner.java (hunk 1 above)
Comment (AnjanaSenanayake) [reply native/health-cli/src/main/java/io/ballerina/health/cmd/Runner.java]: This was added by a mistake. Removed in https://github.com/ballerina-platform/fhir-tools/commit/233fe707741f4680c2b3215341dc823046c19930

File: native/health-cli/src/main/java/io/ballerina/health/cmd/fhir/FhirSubCmd.java (hunk 2 above)
Comment (AnjanaSenanayake) [reply native/health-cli/src/main/java/io/ballerina/health/cmd/fhir/FhirSubCmd.java]: Fixed in https://github.com/ballerina-platform/fhir-tools/commit/12787ad2be0e9593bd17488ebf26924cb41d793b

**Review by isuruh15 — APPROVED** [review]

**Pull request merged**